test/e2e/apps/deployment.go

```diff
@@ -746,6 +750,28 @@ func testIterativeDeployments(f *framework.Framework) {
 	Expect(framework.WaitForDeploymentWithCondition(c, ns, deploymentName, deploymentutil.NewRSAvailableReason, extensions.DeploymentProgressing)).NotTo(HaveOccurred())
 }
 
+func replicaSetHasDesiredReplicas(rsClient extensionsclient.ReplicaSetsGetter, replicaSet *extensions.ReplicaSet) wait.ConditionFunc {
+	desiredGeneration := replicaSet.Generation
+	return func() (bool, error) {
+		rs, err := rsClient.ReplicaSets(replicaSet.Namespace).Get(replicaSet.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		return rs.Status.ObservedGeneration >= desiredGeneration && rs.Status.Replicas == *(rs.Spec.Replicas), nil
+	}
+}
+
+func replicaSetHasSpecificAvailableReplicas(rsClient extensionsclient.ReplicaSetsGetter, replicaSet *extensions.ReplicaSet, specificAvailableReplicas int32) wait.ConditionFunc {
```

Comment:
Move this one too. Something like `WaitForReplicaSetAvailableReplicas`. "Specific" seems redundant as the expected available pods is passed in.

Reply:
+1